### PR TITLE
[WIP] feat(openapi-v2): add decorator for head op

### PR DESCRIPTION
**Pull request by b-admike**
Add sugar function for head operation.

Connect to #691 

## Checklist

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [x] Related API Documentation was updated
- [ ] Affected artifact templates in `packages/cli` were updated (N/A?)
- [ ] Affected example projects in `packages/example-*` were updated (N/A?)

### CODE REVIEW

**Comment by b-admike** [comment]
@slnode test please

File: packages/openapi-v2/test/unit/controller-spec/controller-decorators.test.ts [hunk 1]
```diff
@@ -52,15 +49,11 @@ describe('Routing metadata', () => {
...
-      .withStringResponse()
-      .build();
+    const operationSpec = givenAnOperationSpec();
 
     class MyController {
       @get('/greet', operationSpec)
-      greet() {
-        return 'Hello world!';
-      }
+      greet() {}
```
Comment (jannyHou) [inline packages/openapi-v2/test/unit/controller-spec/controller-decorators.test.ts]: @b-admike I am not quite familiar with these test cases, but I feel the spec for a function that has a return arg is different than the one that doesn't have a return...so I am not sure how the same `openrationSpec` passes for both?

Maybe the code with your change is correct. It depends on does `operationSpec` contains a response object.

File: packages/openapi-v2/test/unit/controller-spec/controller-decorators.test.ts [hunk 2]
```diff
@@ -371,4 +358,10 @@ describe('Routing metadata', () => {
       in: 'query',
     });
   });
+
+  function givenAnOperationSpec() {
+    return anOperationSpec()
+      .withStringResponse()
```
Comment (jannyHou) [inline packages/openapi-v2/test/unit/controller-spec/controller-decorators.test.ts]: hmm...I think if this helper function calls `withStringResponse` then we probably could not omit the `return` in those controller functions.

File: packages/openapi-v2/test/unit/controller-spec/controller-decorators.test.ts (hunk 1 above)
Comment (b-admike) [reply packages/openapi-v2/test/unit/controller-spec/controller-decorators.test.ts]: @jannyHou I think these tests are for asserting that the correct operation object is mapped to its corresponding controller method. Yeah I think the response object in `operationSpec` is the same throughout the tests.

File: packages/openapi-v2/test/unit/controller-spec/controller-decorators.test.ts (hunk 2 above)
Comment (b-admike) [reply packages/openapi-v2/test/unit/controller-spec/controller-decorators.test.ts]: Discussed this with Janny, but for posterity, we do not have any assertions in the tests for the response object or the actual response, and thus having a return statement with a string for the controller methods doesn't have any effect.

**Comment by raymondfeng** [comment]
@b-admike Can you explain why you removed all return values for `greet`?

**Comment by b-admike** [comment]
@raymondfeng it's because we weren't using them in the tests and it increased our code coverage. See my discussion with Janny above.

**Comment by raymondfeng** [comment]
Tests are examples for some developers to learn and follow. I would rather not sacrifice them for artificial coverage numbers. It's better to use `/* istanbul ignore next */` to instruct the tool. See  https://github.com/gotwarlost/istanbul/blob/master/ignoring-code-for-coverage.md.

**Comment by b-admike** [comment]
I mainly got rid of them because I thought they didn't serve any purpose. The coverage increase was just a bonus :). But I don't mind reverting those changes and use the ignore comments.

**Comment by raymondfeng** [comment]
@b-admike It does serve some purposes to show the usage of our decorators and when they apply. For example, `GET` usually expects to see a response from the return value. Developers might be confused to see `greet` does nothing and returns nothing.

**Comment by virkt25** [comment]
Great addition!! Just thinking about this from a developer perspective though ... declaring an additional @head() with a function seems like extra work. Can this decorator be nested on a function doing the @get() call but make the function just return the headers. Or perhaps we should consider a way to mark a GET route as HEAD enabled ... or we just enable it by default for all GET requests and the decorator allows header values to be set? 

I'm ok with the changes. Asking this question more for the purposes of discussion before any changes are made. Reference for purpose of HEAD request: https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/HEAD

**Comment by b-admike** [comment]
@virkt25 Those are good points. I'm up for finding alternatives to this approach! Not sure how to implement what you've suggested though.

**Comment by raymondfeng** [comment]
@virkt25 Good point. 

@b-admike Maybe it's not so useful to have `@head` any more and `@get` will imply. I think `HEAD` can be handled by the router so that requests will be routed to `GET` but remove the response body.

**Comment by virkt25** [comment]
I think it might be better from a performance standpoint to not execute the GET function at all rather than to remove the response body. Unnecessary delay in response to HEAD + additional hits against DB. 

But we should be able to set the content-type header (json / xml / whatever) for the expected response. And for files (if static handler will be using the same router) ... we should be able to return the Content-Length header and type header appropriately as well.

**Comment by raymondfeng** [comment]
Without invoking the get, head won’t have the same list of response headers.

**Comment by virkt25** [comment]
I was just wondering if we may be able to avoid that overhead by letting users optionally specify the headers they want to pass along for a `HEAD` request. But if there's no other way then I'm fine with calling `GET` handler and dropping the body. Seems to be what Express does:

> The app.get() function is automatically called for the HTTP HEAD method in addition to the GET method if app.head() was not called for the path before app.get().

**Comment by raymondfeng** [comment]
openapi-v2 is now removed.

**Pull request closed without merging**